app/(routes)/packets/packet-details.tsx

```diff
@@ -1,23 +1,44 @@
 import Link from 'next/link';
 import { Packet, PacketStates, stateToString } from 'utils/types/packet';
 import { CHAIN_CONFIGS, CHAIN, clientToDisplay } from 'utils/chains/configs';
-import { classNames } from 'utils/functions';
+import { Chain } from 'utils/types/chain';
+import { classNames, numberWithCommas } from 'utils/functions';
 import { CopyButton } from 'components/copy-button';
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
 
 export function PacketDetails(packet: Packet | null) {
-  let sourceUrl = '';
-  let destUrl = '';
-
-  for (const chain of Object.keys(CHAIN_CONFIGS)) {
-    const chainName = chain as CHAIN;
-    const chainVals = CHAIN_CONFIGS[chainName];
-    if (packet?.sourceClient?.toLowerCase().startsWith(chain)) {
-      sourceUrl = chainVals.txUrl;
+  const [sourceChain, setSourceChain] = useState<Chain | undefined>();
+  const [destChain, setDestChain] = useState<Chain | undefined>();
```

Comment:
I am curious why you using undefined instead of null here?

Reply:
Just preference, but I figured since it starts out completely unset it should be undefined rather than null.
https://google.github.io/styleguide/tsguide.html#undefined-and-null